idea/src/org/jetbrains/kotlin/idea/inspections/AnnotationTargetExpressionInspection.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.builtins.KotlinBuiltIns
+import org.jetbrains.kotlin.idea.caches.resolve.resolveToCall
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+
+class AnnotationTargetExpressionInspection : AbstractKotlinInspection() {
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return classVisitor(fun(element: KtClass) {
+            if (!element.isAnnotation()) return
+            val target = element.targetAnnotationArgument()?.asElement() ?: return
+            if (element.retentionAnnotation() == null) return
+
+            val start = element.startOffset
+            val highlightRange = TextRange(target.startOffset - start, target.endOffset - start)
+            holder.registerProblem(
+                holder.manager.createProblemDescriptor(
+                    element,
+                    highlightRange,
+                    "annotation will not be stored",
```

Comment:
Message is not capitalized:

<img width="324" alt="screenshot 2018-11-05 at 19 45 11" src="https://user-images.githubusercontent.com/95996/47993668-64787e00-e133-11e8-853f-f96db23233c7.png">

Reply:
By the way, in Kotlin 1.3. we have compiler warnings/errors for this (`RESTRICTED_RETENTION_FOR_EXPRESSION_ANNOTATION`, `RESTRICTED_RETENTION_FOR_EXPRESSION_ANNOTATION_WARNING`). I think it will make the perfect sense to remove `AnnotationTargetExpressionInspection` and bind `ChangeAnnotationRetentionToSourceFix` directly to them.

Reply:
Please review https://github.com/JetBrains/kotlin/pull/1784. I'll close this PR.